### PR TITLE
Fix replication index conflicts.

### PR DESCRIPTION
**Pull request by danielhlarkin**

### CODE REVIEW

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3892/

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3916/

File: arangod/Replication/Syncer.cpp [hunk 1]
```diff
@@ -748,6 +744,62 @@ Result Syncer::createIndex(VPackSlice const& slice) {
   return Result();
 }
 
+void Syncer::createIndexInternal(VPackSlice const& idxDef, LogicalCollection& col) {
+  std::shared_ptr<arangodb::Index> idx;
+  auto physical = col.getPhysical();
+  TRI_ASSERT(physical != nullptr);
+
+  // check any identifier conflicts first
+  {
```
Comment (graetzer) [inline arangod/Replication/Syncer.cpp]: is it really valid to just drop an index here ?

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/3920/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/3929/

**Pull request merged**